war/src/main/java/io/lumeer/engine/controller/CollectionMetadataFacade.java

```diff
@@ -245,33 +247,35 @@ public void createInitialMetadata(String internalCollectionName, String original
     * @throws AttributeAlreadyExistsException
     *       when attribute with new name already exists
     */
-   public void renameAttribute(String collectionName, String oldName, String newName) throws AttributeAlreadyExistsException {
+   public void renameAttribute(String collectionName, String oldName, String newName) throws AttributeAlreadyExistsException, CollectionMetadataDocumentNotFoundException {
       if (getAttributesNames(collectionName).contains(newName)) {
          throw new AttributeAlreadyExistsException(ErrorMessageBuilder.attributeAlreadyExistsString(newName, collectionName));
       }
 
-      // TODO: does not work and I just can't find out why...
-      //      DataDocument renameQuery = new DataDocument()
-      //            .append("findAndModify", METADATA_COLLECTION)
-      //            .append("query", new DataDocument(LumeerConst.Document.ID, documentId))
-      //            .append("update", new DataDocument()
-      //                  .append("rename", new DataDocument(
-      //                        nestedAttributeName(LumeerConst.Collection.ATTRIBUTES_KEY, oldName),
-      //                        nestedAttributeName(LumeerConst.Collection.ATTRIBUTES_KEY, newName))));
-      //
-      //      dataStorage.run(renameQuery);
-
-      // workaround, big and ugly, but at least works
       DataDocument metadata = getCollectionMetadataDocument(collectionName);
       String documentId = metadata.getId();
-      DataDocument attribute = metadata.getDataDocument(LumeerConst.Collection.ATTRIBUTES_KEY).getDataDocument(oldName);
-      dataStorage.dropAttribute(METADATA_COLLECTION, documentId, nestedAttributeName(LumeerConst.Collection.ATTRIBUTES_KEY, oldName));
-      dataStorage.updateDocument(
-            METADATA_COLLECTION,
-            new DataDocument(
-                  nestedAttributeName(LumeerConst.Collection.ATTRIBUTES_KEY, newName),
-                  attribute),
-            documentId);
+
+      // still does not pass the test, even with $rename
+      DataDocument renameQuery = new DataDocument()
+            .append("findAndModify", METADATA_COLLECTION)
+            .append("query", new DataDocument(LumeerConst.Document.ID, documentId))
```

Comment:
See the code snippet in my previous comment. The following line should be here:
```java
.append("query", new DataDocument(LumeerConst.Collection.INTERNAL_NAME_KEY, collectionName))
```

And the previous getting of metadata document is not necessary.